modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc

```diff
@@ -22,43 +22,153 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
   namespace: openshift-machine-api
 spec:
+  online: True
+  bootMACAddress: <nic1_mac_address> <6>
   bmc:
-    address: <protocol>://<bmc_url> <1>
-    credentialsName: "worker1-bmc"
-  bootMACAddress: <nic1_mac_address>
-  externallyProvisioned: false <2>
-  customDeploy:
+    address: <protocol>://<bmc_url> <7>
+    credentialsName: openshift-worker-<num>-bmc-secret <1>
+    disableCertificateVerification: True <8>
+  rootDeviceHints:
+    deviceName: <root_device_hint> <9>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <1>
+----
++
+--
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name`, `credentialsName`, and `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that has the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces" for details on configuring NMState syntax.
+<3> Optional: If you have configured the network interface with `nmstate`, and you want to disable an interface, set `state: up` with the IP addresses set to `enabled: false` as shown:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: Secret
+metadata:
+ name: openshift-worker-<num>-network-config-secret
+ namespace: openshift-machine-api
+ # ...
+   interfaces:
+   - name: <nic_name>
+     type: ethernet
+     state: up
+     ipv4:
+       enabled: false
+     ipv6:
+       enabled: false
+# ...
+----
+<4> Replace `<nic1_name>`, `<ip_address>`, `<dns_ip_address>`, `<next_hop_ip_address>`, and `<next_hop_nic1_name>` with appropriate values.
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, Redfish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
+<8> Optional: To skip TLS certificate validation, set `disableCertificateVerification` to `true`. Disabling TLS certificate validation is not recommended.
+<9> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint.
+--
+
+* *DHCP configuration* `bmh.yaml`:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openshift-worker-<num>-bmc-secret <1>
+  namespace: openshift-machine-api
+type: Opaque
+data:
+  username: <base64_of_uid> <2>
+  password: <base64_of_pwd> <2>
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: openshift-worker-<num> <1>
+  namespace: openshift-machine-api
+spec:
+  online: True
+  bootMACAddress: <nic1_mac_address> <3>
+  bmc:
+      customDeploy:
```

Comment:
@kdrake20 the indentation and order needs fixing

Reply:
Updated As Suggested.